lib/XML/LibXML/Node.pm

```diff
@@ -108,25 +112,422 @@ role XML::LibXML::Nodish does XML::LibXML::C14N {
+
+        #die "Bad name" if testNodeName($n);
+        #xmlNodeSetName(self, $n);
+
+        if testNodeName($n) {
+            xmlNodeSetName(self.getNode(), $n);        
+        } else {
+            die "Bad name";
+        }
+    }
+
+    method setAttribute(Str $a, Str $v) {
+        sub xmlSetProp(xmlNode, Str, Str)       is native('xml2') { * }
+
+        if testNodeName($a) {
+            # cw: Note, this method locks us into libxml2 versions of 2.6.21 and 
+            #     later. libxml2 does -not- provide us a mechanism to test and 
+            #     implement backwards compatibility.
+            xmlSetProp(self.getNode(), $a, $v);
+        } else {
+            die "Bad name '$a'";
+        }
+    }
+
+    method setAttributeNS(Str $namespace, Str $name, Str  $val) {
+        if ! testNodeName($name) {
+            die "Bad name '$name'";
+            # cw: Yes, I know this looks weird, but is done incase we 
+            #     .return from a CATCH{}
+            return;
+        }
+
+        my ($prefix, $localname) = $name.split(':');
+
+        if !$localname {
+            $localname = $prefix;
+            $prefix := Str;
+        }
+
+        # cw: Sublime currently has crappy syntax highlighting so 
+        #     $namespace ~~ s:g/// will break it. 
+        my xmlNs $ns;
+
+        my $attr_ns;
+        if $namespace.defined {
+            $attr_ns = $namespace.subst(/\s/, '');
+            if $attr_ns.chars {
+                $ns = xmlSearchNsByHref(self.doc, self, $attr_ns);
+
+                if $ns.defined && !$ns.uri {
+                    my @all_ns := nativecast(
+                        CArray[xmlNsPtr],
+                        xmlGetNsList(self.doc, self)
+                    );
+
+                    if (@all_ns.defined) {
+                        my $i = 0;
+                        repeat {
+                            my $nsp := @all_ns[$i++];
+                            $ns = nativecast(xmlNs, $nsp);
+                            last if $ns.uri && ($ns.uri eq $namespace);
+                        } while ($ns);
+                        #xmlFree($all_ns);
+                    }
+                }
+            }
+
+            if (!$ns.defined) {
+                # create new ns
+                if $prefix.defined {
+                    my $attr_p = $prefix.subst(/s/, '');
+                    if $attr_p.chars {
+                        $ns := xmlNewNs(
+                            self.getNode(), $attr_ns, $attr_p
+                        );
+                    } else {
+                        $ns := xmlNs;
+                    }
+                }
+            }
+        }
+
+        if $attr_ns.defined && $attr_ns.chars && ! $ns.defined {
+            die "bad ns attribute!";
+            return            
+        }
+
+        xmlSetNsProp(self.getNode(), $ns, $localname, $val);
+    }
+
+    method hasAttribute($a) {
+        my $ret = domGetAttrNode(
```

Comment:
Yes. In case you haven't noticed, I prefer to read vertically rather than scrolling horizontally. Makes things harder to read. Breaking parameters (especially for long functions) into their own lines also helps. That's one style I will keep in my own code. 

This code is short enough, yes. Originally it was longer.